src/Network/Wai/SAML2.hs

```diff
@@ -114,10 +115,9 @@ saml2Callback cfg callback app req sendResponse = do
                 -- extract the SAML response from the request body
                 [("SAMLResponse",val)] -> do 
                     result <- validateResponse cfg val
-
+                    let rs = lookup "RelayState" body
```

Comment:
This will currently always return `Nothing` because the case alternative for the surrounding case expression only matches when the body is exactly `[("SAMLResponse",val)]` (line 116). I think with this change, the case expression should be:
```haskell
case lookup "SAMLResponse" body of
    Just val -> do 
        -- ...
    Nothing -> callback (Left InvalidRequest) app req sendResponse
```